### ISSUE TITLE
Add ability to use async Jinja2 templates

### ISSUE THREAD

**Issue by pds2208** [tiangolo/fastapi#3129]
I, and I'm sure others, need to use async Jinja2 templates. Primary reason being that I need to add a custom filter that has to be async because it relies on other functions that are async.

Please add this capability.

**Comment by Kludex** [comment]
https://fastapi.tiangolo.com/advanced/templates/?h=jinja2#using-jinja2templates

![image](https://user-images.githubusercontent.com/7353520/115996611-42fbfd00-a5e0-11eb-81f7-a0376ded4d1b.png)

If you want to request some change on this class, you should ask on `Starlette`. FastAPI just reimports that class.

**Comment by tiangolo** [comment]
Yep, what @Kludex said. I don't think Jinja2 itself would support it. So, let me suggest you wrap your async function with Asyncer's `syncify`: https://asyncer.tiangolo.com/

If that solves the original problem, then you can close this issue ✔️

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.

**Comment by Kludex** [comment]
We are studying this on Starlette: https://github.com/encode/starlette/pull/1701
